### PR TITLE
Fix/timelines api migration

### PR DESCRIPTION
**Pull request by apfelfabrik**
Wraps up https://github.com/opf/openproject/pull/527:

> roll-up branch for migrating the rest of the API-views to RABL (given, that we actually make things faster...)

This one also restores some timelines functionality. ~~Unfortunately, Timelines are still missing the capability to fetch users trough their new API, which is why this PR explicitly leaves [`#2577`](https://www.openproject.org/issues/2577) unfixed.~~ Thanks, @nkenneweg.

### CODE REVIEW

File: app/assets/javascripts/timelines.js [hunk 1]
```diff
@@ -721,7 +722,7 @@ Timeline = {
         this.augmentProjectsWithProjectTypesAndAssociations();
 
         this.augmentPlanningElementsWithHistoricalData();
-        this.augmentPlanningElementsWithProjectAndParentAndChildInformation();
+        this.augmentPlanningElementsWithAllKindsOfStuff();
```
Comment (hschink) [inline app/assets/javascripts/timelines.js:730]: Eventually, this will lead to maintenance hell! :fire: Let it in for fun! :laughing:

**Comment by apfelfabrik** [comment]
Merged w/ other changes once more, Changelog is prepared to merge fine w/ release 23.

**Comment by hschink** [comment]
@apfelfabrik, @nkenneweg I'm sorry but tests (cucumber and specs) fail both on Travis and the internal CI. I propose to fix those tests before merging.

**Comment by apfelfabrik** [comment]
Nooooooooooooooooooooo!

File: doc/CHANGELOG.md [hunk 2]
```diff
@@ -38,9 +46,9 @@ See doc/COPYRIGHT.rdoc for more details.
 * `#1916` Work package update screen is closed when attached file is deleted
 * `#1935` Fixed bug: Default submenu for wiki pages is wrong (Configure menu item)
 * `#2009` No journal entry created for attachments if the attachment is added on container creation
-* `#2221` [Accessibility] enhance keyboard shortcuts
 * `#2026` 404 error when letters are entered in category Work Package
 * `#2129` Repository: clicking on file results in 500
+* `#2221` [Accessibility] enhance keyboard shortcuts
```
Comment (nkenneweg) [inline doc/CHANGELOG.md:51]: Der steht oben auch nochmal. Komisch.

**Comment by tilt** [comment]
We could sometime update the routes
-      resources :planning_element_journals
-      resources :statuses
-      resources :planning_element_journals, only: [:index, :create]
-      resources :statuses, only: [:index, :show]
  ...

**Pull request merged**